### DISCUSSION TITLE
Starting applications on specified workspaces failure

### DISCUSSION THREAD

**Discussion by guido99** [i3/i3#4907] [Q&A]
On starting i3, I want to put a text help file on WS 0, google chrome on WS 1, an xterm on WS 2, gv on WS3, and be looking at WS 3 when all is settled.

The relevant snippet from my i3 config:
 
assign [class="XTerm" title="most"] $ws0
assign [class="Google-chrome"] $ws1
exec xterm -e most ~/.config/i3/help
exec google-chrome
exec i3-msg 'workspace 2 ; exec xterm ; workspace 2'
exec i3-msg 'workspace 3 ; exec gv ; workspace 3'

This opens WS 0 and 1 correctly. But, xterm and gv become 2 windows on WS 3. There is no WS 2.

If I comment out / remove the gv launch, it all works as expected. All 3 applications on 3 separate workspaces.

Using assign for xterm and gv works, but, I do **not** want these to always be on fixed workspaces.

Is this possible? Have I coded incorrectly? Thanks for any assistance.

**Comment by Airblader** [comment]
> exec i3-msg 'workspace 2 ; exec xterm ; workspace 2'

For this to work, the application must support startup notifications. If it doesn't, it can't work. I'd blindly guess that xterm doesn't support this, but I definitely don't know.

You have another good option for the startup scenario, which is layout saving & restoring. You'll find docs for that on the website under Docs.

**Reply by guido99** [reply]
Xterm does work! With or without "--no-startup-id" makes no difference.

The weird behavior is in the last 2 lines from above:

exec i3-msg 'workspace 2 ; exec xterm ; workspace 2'
exec i3-msg 'workspace 3 ; exec gv ; workspace 3'

Executing either one line alone does what I expect. Executing both lines does start both but **not** on different WSs.

**Reply by Airblader** [reply]
What I said still stands.

**Reply by guido99** [reply]
This works:

bindsym $W+$C+x exec workspace 2 ; exec gv
bindsym $W+$C+z exec workspace 3 ; exec xpdf

This does not work as intended:

exec i3-msg 'workspace 2 ; exec gv'
exec i3-msg 'workspace 3 ; exec xpdf'

The second scheme is the desired one.

**Reply by guido99** [reply]
Sorry, amended

bindsym $W+$C+x exec workspace 2 ; exec gv ; workspace 2
bindsym $W+$C+z exec workspace 3 ; exec xpdf ; workspace 3
